static/js/components/inputs/DateField.js

```diff
@@ -95,11 +95,11 @@ export default class DateField extends React.Component {
       let validatedDay = Maybe.of(1);
       if (!omitDay) {
         validatedDay = validateDay(newEdit.day);
-        newEdit.day = mstr(validatedDay);
+        newEdit.day = dateForm(mstr(validatedDay), newEdit.day);
```

Comment:
The little comparator function could be made more general too, something like:

```js
const firstIfNumEqual = R.curry((x, y) => Number(x) === y ? x : y);
```

then in `.map` you'd just have:

```js
mstr(validatedDay.map(firstIfNumEqual(newEdit.day)))
```

which would make it a little easier to use the same pattern for months, below.